aiida/backends/tests/cmdline/commands/test_process.py

```diff
@@ -219,6 +220,58 @@ def test_list(self):
             self.assertEqual(len(get_result_lines(result)), 1)
             self.assertEqual(get_result_lines(result)[0], str(self.calcs[0].pk))
 
+    def test_list_worker_slot_warning(self):
+        """
+        Test that the if the number of used worker process slots exceeds a threshold,
+        that the warning message is displayed to the user when running `verdi process list`
+        """
+        from aiida.engine import ProcessState
+
+        # Create 6 WorkFunctionNodes (one for each ProcessState)
+        # Do this 28 times such that we have an additional 168 nodes, of which 84 are 'active' states
+        # Total now is 90 active nodes and 90 inactive nodes.
+        for _ in six.moves.range(28):
+            for state in ProcessState:
+
+                calc = WorkFunctionNode()
+                calc.set_process_state(state)
+
+                # Set the WorkFunctionNode as successful
+                if state == ProcessState.FINISHED:
+                    calc.set_exit_status(0)
+
+                calc.store()
+
+        # Override the call to the circus client to retrieve the number of workers
+        # As we don't have a running circus client, this will normally fail, so here we simulate the
+        # response by redefining the function to get the final value we want.
+        import aiida.cmdline.utils.common
+        aiida.cmdline.utils.common.get_num_workers = lambda: 1
```

Comment:
We should probably reset the original function after the test. I can imagine some other test being used in the future that calls it indirectly, and this will send somebody potentially on a wild goose chase! :)